msprime/demography.py

```diff
@@ -58,6 +59,48 @@ class IncompletePopulationMetadataWarning(UserWarning):
     """
 
 
+@functools.lru_cache(maxsize=16)
+def _build_population_table(demography):
```

Comment:
Wouldn't it be better if this function accepted a list of populations, rather than the whole demography? Then we'll use the cache as long as the populations are fixed (they likely are), regardless of whether the demography is fixed (it likely isn't)